packages/js/settings-editor/src/components/single-select-page-edit/index.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import { Button, SelectControl, Spinner } from '@wordpress/components';
+import { createElement, useCallback, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { store as coreDataStore, Page } from '@wordpress/core-data';
+import { close } from '@wordpress/icons';
+import type { DataFormControlProps } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+import type { DataFormItem } from '../../types';
+import { sanitizeHTML } from '../../utils';
+
+// https://github.com/woocommerce/woocommerce/blob/83a090f70d1f7b07325d9df9bd03fe2f753d4fd4/plugins/woocommerce/includes/admin/class-wc-admin-settings.php#L626-L636
+const PAGE_QUERY_ARGS = {
+	orderby: 'menu_order',
+	order: 'asc',
+	status: [ 'publish', 'private', 'draft' ],
+	per_page: -1,
+	_fields: [ 'id', 'title' ],
+};
+
+const usePages = () => {
+	return useSelect( ( select ) => {
+		const { getEntityRecords, hasFinishedResolution } =
+			select( coreDataStore );
+
+		const args: [ string, string, typeof PAGE_QUERY_ARGS ] = [
+			'postType',
+			'page',
+			PAGE_QUERY_ARGS,
+		];
+
+		return {
+			pages: getEntityRecords( ...args ) as Page[],
+			isLoading: ! hasFinishedResolution( 'getEntityRecords', args ),
+		};
+	}, [] );
+};
+
+type SingleSelectPageEditProps = DataFormControlProps< DataFormItem > & {
+	help?: React.ReactNode;
+	className?: string;
+};
+
+export const SingleSelectPageEdit = ( {
+	data,
+	field,
+	onChange,
+	hideLabelFromVision,
+	help,
+	className,
+}: SingleSelectPageEditProps ) => {
+	const { pages, isLoading } = usePages();
+	const { id } = field;
+
+	// DataForm will automatically use the id as the label if no label is provided so we conditionally set the label to undefined if it matches the id to avoid displaying it.
+	// We should contribute upstream to allow label to be optional.
+	const label =
+		field.label === id ? undefined : (
+			<div
+				dangerouslySetInnerHTML={ {
+					__html: sanitizeHTML( field.label ),
+				} }
+			/>
+		);
+
+	const value = field.getValue( { item: data } ) ?? '';
+
+	const onChangeControl = useCallback(
+		( newValue: string ) =>
+			onChange( {
+				[ id ]: newValue,
+			} ),
+		[ id, onChange ]
+	);
+
+	const options = useMemo(
+		() => [
+			{
+				label: isLoading
+					? __( 'Loading…', 'woocommerce' )
+					: __( 'Select a page…', 'woocommerce' ),
+				value: '',
+			},
```

Comment:
Theses options are selectable. Could we make these a placeholder instead of an actual value?

Reply:
Sure, I've changed it in 83f4c85ddeb97904767510ce6062acc669a16249

![Screenshot 2025-04-02 at 11 49 49](https://github.com/user-attachments/assets/f0fb2679-f948-4ca2-9ff1-3f0a52d8980a)